File: 13-15/src/animals/pack/PackAnimal.java
// PackAnimal.java
package animals.pack;

import animals.Animal;

public class PackAnimal extends Animal {
    public PackAnimal(String name, String birthDate) {
        super(name, birthDate);
    }
}

// Horse.java
package animals.pack;

public class Horse extends PackAnimal {
    public Horse(String name, String birthDate) {
        super(name, birthDate);
    }
}

// Camel.java
package animals.pack;

public class Camel extends PackAnimal {
    public Camel(String name, String birthDate) {
        super(name, birthDate);
    }
}

// Donkey.java
package animals.pack;

public class Donkey extends PackAnimal {
    public Donkey(String name, String birthDate) {
        super(name, birthDate);
    }
}

File: 13-15/C++/src/PackAnimal.cpp
#include "PackAnimal.h"

PackAnimal::PackAnimal(const std::string &name, const std::string &birthDate)
    : Animal(name, birthDate) {}


#include "Horse.h"

Horse::Horse(const std::string &name, const std::string &birthDate)
    : PackAnimal(name, birthDate) {}


#include "Camel.h"

Camel::Camel(const std::string &name, const std::string &birthDate)
    : PackAnimal(name, birthDate) {}


#include "Donkey.h"

Donkey::Donkey(const std::string &name, const std::string &birthDate)
    : PackAnimal(name, birthDate) {}


